VideoLocker/src/main/java/org/edx/mobile/http/OauthRefreshTokenAuthenticator.java

```diff
@@ -118,7 +120,9 @@ private String getErrorCode(String responseBody) {
             JSONObject jsonObj = new JSONObject(responseBody);
             return jsonObj.getString("error_code");
         } catch (JSONException ex) {
-            return null;
+            String error = "Unable to get error_code from 401 response";
+            logger.warn(error);
+            return error;
```

Comment:
It would probably be semantically better to still return `null`, and add a check for it before switching on it.